From dc101c5959402ca53e61423a10c1306b55bc5b6b Mon Sep 17 00:00:00 2001
From: Oleg Karfich <oleg.karfich@wago.com>
Date: Wed, 8 Jul 2020 08:30:23 +0200
Subject: [PATCH] ARM: dts: pfc: dont reset internal watchdog during boot

TI's omap hwmod infrastructure resets all hardware device ip's during
startup. This is ok for all devices except for the watchdog timer
because this one should protect the device from an broken update by
resetting the device. Dont reset the internal watchdog during boot by
adding this ip to the reset-exception table.

Signed-off-by: Oleg Karfich <oleg.karfich@wago.com>
---
 arch/arm/boot/dts/am335x-pfc.dtsi          | 5 +++++
 arch/arm/boot/dts/am3505-pfc-750_820x.dtsi | 5 +++++
 2 files changed, 10 insertions(+)

diff --git a/arch/arm/boot/dts/am335x-pfc.dtsi b/arch/arm/boot/dts/am335x-pfc.dtsi
index 5cdd5de..b0aa66d 100644
--- a/arch/arm/boot/dts/am335x-pfc.dtsi
+++ b/arch/arm/boot/dts/am335x-pfc.dtsi
@@ -320,3 +320,8 @@
 	pinctrl-names = "default";
 	pinctrl-0 = <&gpio3_pins>;
 };
+
+&wdt2 {
+	ti,no-reset-on-init;
+	ti,no-idle;
+};
diff --git a/arch/arm/boot/dts/am3505-pfc-750_820x.dtsi b/arch/arm/boot/dts/am3505-pfc-750_820x.dtsi
index 054c6a6..2b7502b 100644
--- a/arch/arm/boot/dts/am3505-pfc-750_820x.dtsi
+++ b/arch/arm/boot/dts/am3505-pfc-750_820x.dtsi
@@ -684,4 +684,9 @@
 	status = "okay";
 };
 
+&wdt2 {
+	ti,no-reset-on-init;
+	ti,no-idle;
+};
+
 /include/ "pxc-nandparts.dtsi"
-- 
2.7.4

